packages/core/test/guards.test.ts

```diff
@@ -403,3 +408,399 @@ describe('guards - other', () => {
     expect(service.state.value).toBe('c');
   });
 });
+
+describe('guards with child guards', () => {
+  it('guards can contain child guards', () => {
+    const machine = createMachine(
+      {
+        initial: 'a',
+        states: {
+          a: {
+            on: {
+              EVENT: {
+                target: 'b',
+                guard: {
+                  type: 'testGuard',
+                  name: 'any',
+                  children: [
```

Comment:
I'm not super convinced that having a custom predicate that is responsible for resolving its children is a good idea - if u need something like this you can already use `params`, shouldn't `children` just be reserved for combinators like `not`, `and` and `or`?

In other words - could u prepare a real-life example of such a thing used in practice?

Reply:
> if u need something like this you can already use `params`

Not exactly, since `params` are just params - they're not resolved. `children` are resolved.

The usefulness of this is so that custom guards that might need to reference other guards can be made in userland (such as `xor(...)`, etc.).